feat(config): add a config options for healthcheck timeout (#22922)
vectordotdev/vector: changelog.d/22922_healthcheck_timeout_config.feature.md

> Added `timeout` config options for `healthcheck`, to configure timeout duration for healthchecks. Previously it was hardcoded to 10 seconds across all components, but now it can be configured per component.

Reviewer: ```suggestion
Added `timeout` config option to the `healthcheck` sink configuration. Previously it was hardcoded to 10 seconds across all components, but now it can be configured per component.
```
